jar-connector/src/main/java/com/sixsq/slipstream/connector/CliConnectorBase.java

```diff
@@ -416,6 +453,36 @@ static void genAndSetRunApiKey(Run run, User user, Map<String, String> environme
 		setApiKeyOnRun(run, apiKeySecret.get(API_KEY_KEY));
 	}
 
+	static List<String> extractDeploymentCredentialIds(Run run, User user) {
+
+		List<String> deploymentCredentials = new ArrayList<String>();
+
+		// Get the list of cloud services being used for the deployment
+		String[] cloudServiceNames = run.getCloudServiceNamesList();
+
+		Set<String> cloudServiceNamesSet = new HashSet<String>();
+		for (String cloudServiceName : cloudServiceNames) {
+			cloudServiceNamesSet.add("connector/" + cloudServiceName);
+		}
+
+		// Get the credentials visible to the user
+		CloudCredentialCollection cloudCredentials = User.findCloudCredentials(user);
+
+		for (CloudCredential cloudCredential : cloudCredentials.getCredentials()) {
+			if (cloudServiceNamesSet.contains(cloudCredential.href)) {
```

Comment:
It should be `cloudCredential.connector.href` (`href` is public on `connector`).  Or it can even be:
```
cloudServiceNames.contains(cloudCredential.connector.getRefResourceName())
```
i.e. w/o needing to build `cloudServiceNamesSet` out of resource IDs, because `getRefResourceName()` returns only the name part.  
See https://github.com/slipstream/SlipStreamServer/blob/26dfd688bc8f6113e979815f2c64cf750a7b0eaa/jar-persistence/src/main/java/com/sixsq/slipstream/credentials/HRef.java